jax/numpy/linalg.py

```diff
@@ -86,9 +86,11 @@ def slogdet(a):
 
 
 @_wraps(onp.linalg.det)
+@custom_transforms
 def det(a):
   sign, logdet = slogdet(a)
   return sign * np.exp(logdet)
+defjvp(det, lambda g, ans, x: np.trace(np.linalg.solve(x, g))*ans)
```

Comment:
will this re-use the LU decomposition from the forward pass of `slogdet`?

Reply:
No - it would just run `solve` directly on the primal and tangent. In this case it might actually be faster to not do a custom gradient, though in general, I do wonder if it's possible to define a jvp that closes over variables in the scope of the primal computation, like you can do with `@custom_gradient` in tensorflow?

Reply:
JAX has `custom_gradient`, too. It might be worth giving that a try.

Reply:
True. I thought that was just syntactic sugar though, and this was the recommended way to overload derivatives.

Reply:
I think using `custom_transforms` + `defjvp` is indeed the better way for this case, since `custom_gradient` only sets up a reverse-mode rule whereas here we're getting both a forward- and reverse-mode rule.

As for reusing the factorization, we could make that work, though this current PR is already a big improvement, and under a `jit` XLA will likely CSE things to reuse the factorization anyway. To make it save work without a `jit`, we can make a follow-up PR.

Reply:
Right, we could use `defjvp_all` to reuse the factorization while still defining a JVP rule.

Agreed about saving that for a follow-up PR, though...